[export] Fix cond for pass_base

angelayi opened the pull request:
I ported over the code for the inline interpreter incorrectly in the pass base 😅 

Originally the function `make_inline_interpreter` is supposed to take in a fx.Interpreter type but I accidentally passed in an fx.Interpreter object. Also realized while modifying this diff (and comments from Tugsuu) that we don't really need this InlineInterpreter.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100836

angelayi committed:
[export] Fix cond for pass_base

[ghstack-poisoned]

tugsbayasgalan reviewed (commented):
Sorry I am not super familiar with ExportPass, could you provide a summary of what you changed and how it fixed the problem?

tugsbayasgalan commented on torch/_export/pass_base.py:
```diff
@@ -264,12 +259,11 @@ def _fx(
             ProxyValue, lambda x: x.data, (args, kwargs)
         )
-        res_data = getattr(interpreter, kind)(target, args_data, kwargs_data)
+        res_data = getattr(InlineInterpreter(), kind)(target, args_data, kwargs_data)
```
Previously it seems like we used to reuse interpreter, but now we are creating a new one each time we invoke fx operator. Is it intended?

angelayi replied:
You're right -- sorry, I don't think we need the InlineInterpreter anymore...will just get rid of it and use the interpreter that we already have 😂

angelayi committed:
Update on "[export] Fix cond for pass_base"

I ported over the code for the inline interpreter incorrectly in the pass base 😅 

Originally the function `make_inline_interpreter` is supposed to take in a fx.Interpreter type but I accidentally passed in an fx.Interpreter object.

Context on what the InlineInterpreter does: its a way for us to get the result of running a specific node in the graph with the arguments available at the specific time so that we can get the "val" metadata associated with that node. Let's say we want to add the node `call_function(op=add, args=(x, y), kwargs={})`. At that point we have fake tensors associated with `x` and `y`, and we want to get the result of running `add(x, y)` so that we can set the `val` metadata for the new call_function node. So we run this call_function node with the arguments which basically just calls `add(x, y)` and returns us the result.




[ghstack-poisoned]

angelayi commented:
@pytorchbot merge

angelayi commented:
@pytorchbot merge
